sources/packages/backend/apps/api/src/route-controllers/assessment/_tests_/e2e/assessment.institutions.controller.getAssessmentNOA.e2e-spec.ts

```diff
@@ -0,0 +1,270 @@
+    db = createE2EDataSources(dataSource);
+    // College F.
+    const { institution: collegeF } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeFLocation = createFakeInstitutionLocation(collegeF);
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+  });
+
+  it("Should get the student noa details for an eligible application when an eligible public institution user tries to access it.", async () => {
+    // Arrange
+
+    // Student has an application to the institution eligible for NOA.
+    const student = await saveFakeStudent(db.dataSource);
+
+    const currentMSFAA = createFakeMSFAANumber(
+      { student },
+      {
+        state: MSFAAStates.Signed,
+      },
+    );
+    await db.msfaaNumber.save(currentMSFAA);
+    const application = await saveFakeApplicationDisbursements(db.dataSource, {
+      institution: collegeF,
+      institutionLocation: collegeFLocation,
+      student,
+      msfaaNumber: currentMSFAA,
+    });
+    const assessment = application.currentAssessment;
+    const [firstDisbursementSchedule] =
+      application.currentAssessment.disbursementSchedules;
+
+    const awards = {};
+    firstDisbursementSchedule.disbursementValues.forEach((disbursement) => {
+      awards[`disbursement1${disbursement.valueCode.toLowerCase()}`] =
+        disbursement.valueAmount;
+    });
+
+    const endpoint = `/institutions/assessment/student/${student.id}/assessment/${assessment.id}/noa`;
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        applicationId: application.id,
+        applicationNumber: application.applicationNumber,
+        applicationStatus: application.applicationStatus,
+        assessment: assessment.assessmentData,
+        disbursement: {
+          disbursement1COEStatus: firstDisbursementSchedule.coeStatus,
+          disbursement1Date: getDateOnlyFormat(
+            firstDisbursementSchedule.disbursementDate,
+          ),
+          disbursement1Id: firstDisbursementSchedule.id,
+          disbursement1MSFAACancelledDate:
+            firstDisbursementSchedule.msfaaNumber?.cancelledDate,
+          disbursement1MSFAADateSigned:
+            firstDisbursementSchedule.msfaaNumber?.dateSigned,
+          disbursement1MSFAAId: firstDisbursementSchedule.msfaaNumber?.id,
+          disbursement1MSFAANumber: MASKED_MSFAA_NUMBER,
+          disbursement1Status:
+            firstDisbursementSchedule.disbursementScheduleStatus,
+          disbursement1TuitionRemittance:
+            firstDisbursementSchedule.tuitionRemittanceRequestedAmount,
+          ...("disbursement1bcsl" in awards && {
```

Comment:
This would be the same from https://github.com/bcgov/SIMS/pull/1952#discussion_r1207334529, right?